cmake/toolchain/gnuarmemb/generic.cmake

```diff
@@ -24,6 +24,7 @@ set(BINTOOLS gnu)
 
 set(CROSS_COMPILE_TARGET arm-none-eabi)
 set(SYSROOT_TARGET       arm-none-eabi)
+set(CMAKE_EXE_LINKER_FLAGS_INIT "--specs=nosys.specs")
```

Comment:
Let alone the problem that this is trying to solve does not exist when using one of the supported configurations (see https://github.com/zephyrproject-rtos/zephyr/issues/45383#issuecomment-1118672615), this fix is wrong because `--specs=nosys.specs` is only valid when using the minimal libc. When using the newlib-nano, `--specs=nano.specs` is used.